### PR TITLE
Remove unnecessary net compatibility workarounds

### PR DESCRIPTION
**Pull request by kinkie**
All modern network stacks provide inet_ntop(), inet_pton(),
getaddrinfo(), and getnameinfo() APIs.

Also removed related Solaris 9 hacks, effectively discontinuing support
for that platform. Solaris 9 is no longer supported by Oracle.

### CODE REVIEW

File: configure.ac
Comment (yadij) [inline configure.ac:2721]: I have been reviewing the Sockets API fro MinGW recently and it seems modern Windows API (Vista+) is included in the set which provides the POSIX syscalls. I think we can drop these ```InetPtonA``` and ```InetNtopA``` as well.

File: compat/Makefile.am
Comment (rousskov) [inline compat/Makefile.am:32]: I have adjusted PR title/description. I started that edit to fix M-failed-description, but then discovered that the PR removes more than "functions" and edited some more. Please review and adjust further as needed.

> Adjusted PR description: All modern network stacks provide inet_ntop(), inet_pton(),
getaddrinfo(), and getnameinfo() APIs.

The code remaining in compat/os/opensolaris_10_netdb.h still defines `struct addrinfo`, `EAI_OVERFLOW`, and various other getaddrinfo() API components (see commit 3729717). PR description can be interpreted as implying that Squid only supports "modern network stacks". Based on that assumption, we should remove all those Solaris replacements, possibly undoing the entire hack introduced in commit 3729717 and, hence, effectively dropping Solaris 9 support (if it is still considered supported by anyone).

I do not insist on any changes.

**Review by rousskov — COMMENTED** [review]
Thank you for this cleanup. I hope your assumption regarding modern TCP/IP stacks is correct and relevant to all Squid platforms we want to support.

**Comment by kinkie** [comment]
> Thank you for this cleanup. I hope your assumption regarding modern TCP/IP stacks is correct and relevant to all Squid platforms we want to support.

I checked the manual pages of the environments we support, I can try a build on freebsd and openbsd before merge; mingw (and therefore windows) will be double checked as part of the ongoing efforts there

File: compat/Makefile.am
Comment (kinkie) [reply compat/Makefile.am:32]: Solaris 9 is not supported by Oracle anymore; I'll remove

File: configure.ac
Comment (kinkie) [reply configure.ac:2721]: removed

File: compat/Makefile.am
Comment (kinkie) [reply compat/Makefile.am:32]: Removed

**Comment by rousskov** [comment]
> I can try a build on freebsd and openbsd before merge

Your call.

File: compat/Makefile.am
Comment (rousskov) [reply compat/Makefile.am:32]: Glad those hacks can be removed now. I added the corresponding note to the PR description.

**Comment by kinkie** [comment]
> > I can try a build on freebsd and openbsd before merge
> 
> Your call.

https://build.squid-cache.org/job/anybranch-openbsd-matrix/17/COMPILER=clang,label=openbsd/console

**Comment by kinkie** [comment]
> > > I can try a build on freebsd and openbsd before merge
> > 
> > 
> > Your call.
> 
> https://build.squid-cache.org/job/anybranch-openbsd-matrix/17/COMPILER=clang,label=openbsd/console

Test failed but due to an issue with the testing framework, the code built fine.

**Review by rousskov — APPROVED** [review]
Thank you for this cleanup!

Disclaimer: I cannot verify the claims in the PR description.

**Comment by squidadm** [comment]
queued for backport to v6

**Pull request closed without merging**
